spec/_plugins/locale_url_filter_spec.rb

```diff
@@ -0,0 +1,278 @@
+          let(:locale) { 'es' }
+
+          it 'renders without locale' do
+            expect(locale_url).to eq('/es/path/')
+          end
+        end
+      end
+    end
+
+    context 'with site base url' do
+      context 'without collection in permalink' do
+        let(:config) do
+          {
+            'baseurl' => 'https://example.com',
+            'collections' => {
+              'en' => { 'permalink' => '/:path/' },
+              'es' => { 'permalink' => '/:path/' },
+            },
+          }
+        end
+
+        context 'without path' do
+          context 'without locale' do
+            it 'renders without locale' do
+              expect(locale_url).to eq('https://example.com/')
+            end
+          end
+
+          context 'with locale' do
+            let(:locale) { 'es' }
+
+            it 'renders without locale' do
+              expect(locale_url).to eq('https://example.com/')
+            end
+          end
+        end
+
+        context 'with path' do
+          let(:path) { '/path/' }
+
+          context 'without locale' do
+            it 'renders without locale' do
+              expect(locale_url).to eq('https://example.com/path/')
+            end
+          end
+
+          context 'with locale' do
+            let(:locale) { 'es' }
+
+            it 'renders without locale' do
+              expect(locale_url).to eq('https://example.com/path/')
+            end
+          end
+        end
+      end
+
+      context 'with collection in permalink' do
+        let(:config) do
+          {
+            'baseurl' => 'https://example.com',
+            'collections' => {
+              'en' => { 'permalink' => '/:collection/:path/' },
+              'es' => { 'permalink' => '/:collection/:path/' },
+            },
+          }
+        end
+
+        context 'without path' do
+          context 'without locale' do
+            it 'renders with locale from page data' do
+              expect(locale_url).to eq('https://example.com/en/')
+            end
+          end
+
+          context 'with locale' do
+            let(:locale) { 'es' }
+
+            it 'renders with locale' do
+              expect(locale_url).to eq('https://example.com/es/')
+            end
+          end
+        end
+
+        context 'with path' do
+          let(:path) { '/path/' }
+
+          context 'without locale' do
+            it 'renders without locale' do
+              expect(locale_url).to eq('https://example.com/en/path/')
+            end
+          end
+
+          context 'with locale' do
+            let(:locale) { 'es' }
+
+            it 'renders without locale' do
+              expect(locale_url).to eq('https://example.com/es/path/')
+            end
+          end
+        end
+      end
+    end
+  end
+
+  describe '#delocalize_url' do
+    let(:path) { '' }
+    let(:locale) { nil }
+    let(:page_data) { { 'lang' => 'es' } }
+    subject(:locale_url) { instance.delocalize_url(path, locale) }
```

Comment:
I would consider calling this one something to match the method it's testing?

```suggestion
    subject(:delocalized_url) { instance.delocalize_url(path, locale) }
```

Reply:
Oops, I'd carelessly did this revision as a mass replace. Will update to name appropriately.

Reply:
> I would consider calling this one something to match the method it's testing?

Renamed `delocalized_url` in rebased b7d1c0b.